### PR TITLE
[fuchsia] Use a core-jit snapshot.

### PR DESCRIPTION
**Pull request by rmacnak-google**

### CODE REVIEW

File: lib/snapshot/BUILD.gn [hunk 1]
```diff
@@ -33,9 +68,13 @@ action("generate_snapshot_bin") {
     ":generate_dart_ui",
     "//dart/runtime/bin:gen_snapshot($dart_host_toolchain)",
   ]
+  if (target_is_fuchsia) {
+    deps += [ ":generate_package_map" ]
+  }
   inputs = [
              "//dart/runtime/tools/create_snapshot_bin.py",
              snapshot_dart,
+             "compilation_trace.txt"
```
Comment (zanderso) [inline lib/snapshot/BUILD.gn]: Should this be an input only if target_is_fuchsia?

File: lib/snapshot/BUILD.gn [hunk 2]
```diff
@@ -84,6 +127,10 @@ action("generate_snapshot_bin") {
     args += [
       "--url_mapping=dart:fidl.internal,$fidl_internal_path",
       "--url_mapping=dart:mozart.internal,$mozart_internal_path",
+      "--packages",
+      rebase_path("$target_gen_dir/snapshot.packages"),
+      "--load_compilation_trace",
+      rebase_path("compilation_trace.txt"),
```
Comment (zanderso) [inline lib/snapshot/BUILD.gn]: Maybe fuchsia_compilation_trace.txt

File: lib/snapshot/BUILD.gn (hunk 1 above)
Comment (rmacnak-google) [inline lib/snapshot/BUILD.gn]: Yes, done.

File: lib/snapshot/BUILD.gn (hunk 2 above)
Comment (rmacnak-google) [inline lib/snapshot/BUILD.gn]: That's better. Done.

File: lib/snapshot/snapshot_fuchsia.dart [hunk 3]
```diff
@@ -14,3 +16,22 @@ import 'dart:math';
 import 'dart:mozart.internal';
 import 'dart:typed_data';
 import 'dart:ui';
+
+// These libraries are optional. They are included in the core snapshot and
```
Comment (zanderso) [inline lib/snapshot/snapshot_fuchsia.dart]: Maybe not in this CL, but we should consider adding a README.md in this directory explaining how to create/consume a snapshot for a different app profile.

**Comment by Hixie** [comment]
What is the status of this PR?

**Comment by rmacnak-google** [comment]
@zanderso Updated to only use core-jit when target_cpu == host_cpu.

**Pull request merged**